### DISCUSSION TITLE
Change task interval during execution

### DISCUSSION THREAD

**Discussion by dasrocky** [arkhipenko/TaskScheduler#158] [Q&A] [answered]
Hello,
Is it possible to change the interval of tasks during or in middle of execution? By that i mean:
```
#include <TaskScheduler.h>
int interval = 5000;
Scheduler taskManager;
Task testTask(interval, Task_Forever, &someCallback, &taskManager, true);
void setup(){}
void loop(){
taskManager.execute();
}
```
Somewhere in execution i would like to change the interval, is it best way to change the interval by reintialize(interval = 10000) or is there a better way to change the interval of task "testTask". Reason to change the interval of task is to update/change exectution of task interval like printing or publishing data through mqtt at different intervals rather than being fixed.

Just recently came across this library, and it is one of the best arduino & esp boards libraries out there, thank you.

Thank you in advance.

**Comment by arkhipenko** [comment, marked as answer]
Yes, please read the Wiki (paying specific attention to **NOTES**)

https://github.com/arkhipenko/TaskScheduler/wiki/API-Task#void-setinterval-unsigned-long-ainterval

and an example:
https://github.com/arkhipenko/TaskScheduler/blob/4ccca1ac3d63a2b6fa853df3c93147fa4e8980da/examples/Scheduler_example00_Blink/Scheduler_example00_Blink.ino#L284


You can set it in the Task's own callback method, or outside of the task code, in another task's callback method.
